### PR TITLE
add pytorch-lightning DDP example

### PR DESCRIPTION
**Pull request by tohmae**
This PR addresses issue #14.
This PR is related to optuna/optuna#2849.

### CODE REVIEW

File: pytorch/pytorch_lightning_ddp.py
Comment (nzw0301) [inline pytorch/pytorch_lightning_ddp.py:10]:
```suggestion
    $ python pytorch/pytorch_lightning_ddp.py [--pruning]
```

File: pytorch/pytorch_lightning_ddp.py
Comment (nzw0301) [inline pytorch/pytorch_lightning_ddp.py:2]:
```suggestion
Optuna example that optimizes multi-layer perceptrons using PyTorch Lightning's 
distributed data-parallel training.
```

**Review by nzw0301 — COMMENTED** [review]
Could you update the comments?

**Comment by tohmae** [comment]
Thank you for your comments.
I updated the comment.

**Comment by nzw0301** [comment]
I think this PR has been blocked by https://github.com/optuna/optuna/pull/3157; after merging https://github.com/optuna/optuna/pull/3157, we can review this PR.

**Comment by nzw0301** [comment]
We need to merge https://github.com/optuna/optuna-examples/pull/74 before reviewing this PR.

**Comment by tohmae** [comment]
Thank you for your comments.
I added support for pytorch-lightning>=1.5.0.

**Comment by nzw0301** [comment]
Awesome! I'm not sure why #74's changes in `requirements.txt` are ignored. Could you merge the main branch once?

File: pytorch/pytorch_lightning_ddp.py
Comment (nzw0301) [inline pytorch/pytorch_lightning_ddp.py:146]: how about this?
```suggestion
    parser = argparse.ArgumentParser(
        description="PyTorch Lightning distributed data-parallel training example."
    )
```

**Comment by tohmae** [comment]
Thank you for your comments.
I merged the main branch.

File: .github/workflows/pytorch.yml
Comment (nzw0301) [inline .github/workflows/pytorch.yml:69]: The pruning version seems enough to be tested with the CI. In addition, it might be faster than without pruning. However, I think other callback integrations are tested without the pruning option. So, I'd like to hear the other dev members opinion...

**Review by nzw0301 — APPROVED** [review]

File: .github/workflows/pytorch.yml
Comment (hvy) [reply .github/workflows/pytorch.yml:69]: I agree that it's open for discussion. The total code coverage, including the code path in Optuna, would be improved with pruning turned on. The code coverage inside the example would basically be unchanged thought, and the integration modules are tested in Optuna's main repository. This might be the reasons for the current test configurations in this repository. How about discussing that in a different issue?

Comment (nzw0301) [reply .github/workflows/pytorch.yml:69]: Thank you for your comment! Sounds reasonable. I'll merge this PR as is the current version and I'll create an issue to discuss it.

**Pull request merged**
